nodes/src/nodes/tool_shell/shell_executor.py

```
@@ -0,0 +1,123 @@
+    timeout: int,
+    max_output_bytes: int,
+) -> dict:
+    """Run *command* in the host shell and capture its result.
+
+    Returns a dict with ``stdout``, ``stderr``, ``exit_code``, ``timed_out``,
+    and ``truncated`` keys. Output is decoded as UTF-8 (errors replaced) and
+    capped at ``max_output_bytes`` bytes for each stream.
+    """
+    timed_out = False
+    try:
+        completed = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)